assets/js/blocks/product-category/block.js

```diff
@@ -48,6 +53,36 @@ class ProductByCategoryBlock extends Component {
 		}
 	}
 
+	startEditing() {
+		this.setState( {
+			isEditing: true,
+			changedAttributes: {},
+		} );
+	}
+
+	stopEditing() {
+		this.setState( {
+			isEditing: false,
+			changedAttributes: null,
+		} );
+	}
+
+	setAttributes( attributes ) {
+		this.setState( ( prevState ) => {
+			if ( prevState.changedAttributes !== null ) {
```

Comment:
I think this is preventing attributes from being set when the block is first added? At least, if I add the block, then try to select categories, they don't select.

Ah, I think it's because `changedAttributes` is `null` until you `startEditing` (and then it's `{}`), but `startEditing` is not called when the block is first added to the page.